### PR TITLE
fix: fox page warnings

### PR DESCRIPTION
**Pull request by chxash**
## Description

Fixes 2 warnings related to recent fox page code.

Existing warnings for fox page on develop branch are :
warning1
warning2

## Notice

- [X] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [X] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Coming from review comment https://github.com/shapeshift/web/pull/1879#pullrequestreview-987068820

## Risk

Risk is the change of tab on fox page is not working properly anymore. I tested it and it's working fine.

## Testing

- Go to fox page
- Verify tab change is still working fine
- No more warnings related to fox page in console

## Screenshots (if applicable)

fox

foxy

### CODE REVIEW

File: src/plugins/foxPage/foxPage.tsx [hunk 1]
```diff
@@ -133,9 +133,9 @@ export const FoxPage = (props: FoxPageProps) => {
             {isLargerThanMd &&
               assets.map((asset, index) => (
                 <FoxTab
+                  key={index}
```
Comment (0xApotheosis) [inline src/plugins/foxPage/foxPage.tsx:136]:
```suggestion
                  key={asset.assetId}
```

File: src/plugins/foxPage/foxPage.tsx [hunk 2]
```diff
@@ -164,11 +164,10 @@ export const FoxPage = (props: FoxPageProps) => {
                   </MenuButton>
                   <MenuList>
                     {assets.map((asset, index) => (
-                      <MenuItem onClick={() => handleTabClick(asset.assetId)}>
+                      <MenuItem key={index} onClick={() => handleTabClick(asset.assetId)}>
```
Comment (0xApotheosis) [inline src/plugins/foxPage/foxPage.tsx:167]:
```suggestion
                      <MenuItem key={asset.assetId} onClick={() => handleTabClick(asset.assetId)}>
```

**Review by 0xApotheosis — CHANGES_REQUESTED** [review]
Thanks for this, @chxash!

Two suggested changes, we should use stable IDs over item indexes whenever possible. 
See here for an explanation: https://reactjs.org/docs/lists-and-keys.html#keys

**Review by 0xApotheosis — APPROVED** [review]

File: src/plugins/foxPage/components/FoxTab.tsx [hunk 3]
```diff
@@ -18,11 +17,11 @@ export const FoxTab: React.FC<FoxTabProps> = ({
   assetSymbol,
   fiatAmount,
   cryptoAmount,
-  isSelected,
   onClick,
   ...props
 }) => {
   const bgHover = useColorModeValue('gray.100', 'gray.750')
+  const { isSelected, ...otherProps } = props
```
Comment (gomesalexandre) [inline src/plugins/foxPage/components/FoxTab.tsx:24]: This doesn't produce warnings anymore, but not for the reason one would initially think. The reason it works is because we're not destructuring `isSelected` from the functional component arguments at runtime, but from the rest `...props`.
Using the [rest parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters) syntax effectively creates a new reference, so all react sanity checks are effectively gone here.

File: src/plugins/foxPage/foxPage.tsx [hunk 4]
```diff
@@ -133,9 +133,9 @@ export const FoxPage = (props: FoxPageProps) => {
             {isLargerThanMd &&
               assets.map((asset, index) => (
                 <FoxTab
+                  key={asset.assetId}
                   assetSymbol={asset.symbol}
                   assetIcon={asset.icon}
-                  isSelected={props.activeAssetId === asset.assetId}
```
Comment (gomesalexandre) [inline src/plugins/foxPage/foxPage.tsx:138]: We're not passing `isSelected` here anymore so there is no reason to handle it in `<FoxTab />`.

File: src/plugins/foxPage/components/FoxTab.tsx [hunk 5]
```diff
@@ -39,9 +38,8 @@ export const FoxTab: React.FC<FoxTabProps> = ({
       _hover={{ textDecoration: 'none', bg: { base: 'none', md: bgHover } }}
       textAlign='left'
       p={0}
-      isSelected={isSelected}
       onClick={onClick}
-      {...props}
+      {...otherProps}
```
Comment (gomesalexandre) [inline src/plugins/foxPage/components/FoxTab.tsx:42]: We're not passing `isSelected` anymore, so why pass it to `<Tab />` / destructure it in this component?
Not only is it not passed, but it is also undefined.

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
See comments, this removes warnings for a cascade of incidental reasons, but either way, it is a non-solution to a non-problem, see https://github.com/chakra-ui/chakra-ui/discussions/5425

`isSelected` was erroneously added in 9fddef83d but chakra's typings are wrong, `isSelected` doesn't do anything. You can demonstrate this by removing all `isSelected` lines altogether (effectively the same as you already did by not passing it down) and the selection will still work: the way the tab selection works for us is because we're using `<Tabs />` in a controlled way:
https://github.com/shapeshift/web/blob/4f9dc3d5b07cc725a8aa063aea3a28afa8910c7f/src/plugins/foxPage/foxPage.tsx#L124

tl;dr the `isSelected` removals can stay, but the rest props destructuring should be reverted.

File: src/plugins/foxPage/components/FoxTab.tsx (hunk 5 above)
Comment (chxash) [reply src/plugins/foxPage/components/FoxTab.tsx]: Fixed ! indeed it is totally useless now. Convinced myself that it was still necessary but I probably didn't rebuild correctly the app locally while doing this.

File: src/plugins/foxPage/components/FoxTab.tsx (hunk 3 above)
Comment (chxash) [reply src/plugins/foxPage/components/FoxTab.tsx]: Agreed !

**Comment by chxash** [comment]
> See comments, this removes warnings for a cascade of incidental reasons, but either way, it is a non-solution to a non-problem, see [chakra-ui/chakra-ui#5425](https://github.com/chakra-ui/chakra-ui/discussions/5425)
> 
> `isSelected` was erroneously added in [9fddef8](https://github.com/shapeshift/web/commit/9fddef83ded33827acd4a6b2c0508670c80d0c23) but chakra's typings are wrong, `isSelected` doesn't do anything. You can demonstrate this by removing all `isSelected` lines altogether (effectively the same as you already did by not passing it down) and the selection will still work: the way the tab selection works for us is because we're using `<Tabs />` in a controlled way:
> 
> https://github.com/shapeshift/web/blob/4f9dc3d5b07cc725a8aa063aea3a28afa8910c7f/src/plugins/foxPage/foxPage.tsx#L124
> 
> tl;dr the `isSelected` removals can stay, but the rest props destructuring should be reverted.

Agree with this, reverted the props destructuring.

**Review by gomesalexandre — APPROVED** [review]
Looking good now! Retested and warning are still gone. Well done ser 🍬

**Review by theoboldfrazier — APPROVED** [review]
